app/models/tagging.rb

```diff
@@ -0,0 +1,4 @@
+class Tagging < ActiveRecord::Base
+  belongs_to :tag
+  belongs_to :post
```

Comment:
Add   validates :tag_id, uniqueness: {scope: :post_id}
so that the Tagging table would only contain unique tag with post.